esrally/telemetry.py

```diff
@@ -664,36 +662,42 @@ def __str__(self):
 
     def record(self):
         """
-        Collect _cat/shards for indexes (optionally) specified in telemetry parameters and push to metrics store.
+        Collect node-stats?level=shards for indexes (optionally) specified in telemetry parameters and push to metrics store.
         """
         # pylint: disable=import-outside-toplevel
         import elasticsearch
         try:
-            stats = self.client.cat.shards(index=self.indices)
+            sample = self.client.nodes.stats(metric="_all", level="shards")
         except elasticsearch.TransportError:
-            msg = "A transport error occurred while collecting _cat/shards on cluster [{}]".format(self.cluster_name)
+            msg = f"A transport error occurred while collecting shard stats on cluster [{self.cluster_name}]"
             self.logger.exception(msg)
             raise exceptions.RallyError(msg)
 
-        # _cat/shards gives out put
-        # index shard prirep state docs store ip node
+        shard_metadata = {
+                "cluster": self.cluster_name
+            }
 
+        for node_stats in sample["nodes"].values():
+            node_name = node_stats["name"]
+            collected_node_stats = collections.OrderedDict()
+            collected_node_stats["name"] = "shard-stats"
+            shard_stats = node_stats["indices"].get("shards")
+
+            for index_name, stats in shard_stats.items():
+                for curr_shard in stats:
+                    for shard_id, curr_stats in curr_shard.items():
```

Comment:
Can we make this more robust in case of missing properties? A benchmark should not fail because a property wasn't contained in the response.